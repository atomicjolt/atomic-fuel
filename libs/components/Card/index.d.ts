import React from 'react';
import './styles.css';
export interface Props {
    classOverride?: string;
    classes?: string;
    title: string;
    subtitle?: string;
    blank?: boolean;
    children?: React.ReactNode;
}
export default function Card(props: Props): JSX.Element;
