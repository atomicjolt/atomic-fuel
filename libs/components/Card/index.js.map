{"version":3,"file":"index.js","sourceRoot":"","sources":["../../../src/components/Card/index.tsx"],"names":[],"mappings":";;;;;AAAA,gDAA0B;AAC1B,0DAA4B;AAE5B,wBAAsB;AAWtB,SAAwB,IAAI,CAAC,KAAY;IAErC,IAAA,aAAa,GAMX,KAAK,cANM,EACb,OAAO,GAKL,KAAK,QALA,EACP,KAAK,GAIH,KAAK,MAJF,EACL,QAAQ,GAGN,KAAK,SAHC,EACR,KAAK,GAEH,KAAK,MAFF,EACL,QAAQ,GACN,KAAK,SADC,CACA;IAEV,IAAM,SAAS,GAAG,aAAa,IAAI,SAAS,CAAC;IAE7C,OAAO,CACL,uCAAK,SAAS,EAAE,IAAA,oBAAE,EAAC,SAAS,EAAE,OAAO,CAAC;QAClC,CAAC,KAAK;YACN;gBACE,uCAAK,SAAS,EAAE,UAAG,SAAS,cAAW;oBACrC,sCAAI,SAAS,EAAE,UAAG,SAAS,oBAAiB,IAAG,KAAK,CAAM;oBAC1D,sCAAI,SAAS,EAAE,UAAG,SAAS,uBAAoB,IAAG,QAAQ,CAAM,CAC5D;gBACN,2CAAS,SAAS,EAAE,UAAG,SAAS,cAAW,IACxC,QAAQ,CACD,CACT;QACH,KAAK,IAAI,8DAAG,QAAQ,CAAI,CACtB,CACP,CAAC;AACJ,CAAC;AA3BD,uBA2BC"}