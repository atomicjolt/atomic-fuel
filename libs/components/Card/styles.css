.aj-card {
  border-radius: 5px;
  box-shadow: 0px 5px 11px -3px rgba(0, 0, 0, 0.36);
  -webkit-box-shadow: 0px 5px 11px -3px rgba(0, 0, 0, 0.36);
  padding: 1.5rem 2rem;
  max-width: 420px;
  font-family: Lato, sans-serif;
  box-sizing: border-box;
  border: 1px solid rgba(0, 0, 0, 0.036); }
  .aj-card__heading {
    display: flex;
    flex-direction: column;
    align-items: baseline;
    justify-content: space-between;
    padding-bottom: 0.25rem; }
    .aj-card__heading > h1,
    .aj-card__heading > h2 {
      padding: 0;
      margin: 0; }
    .aj-card__heading-title {
      font-size: 1.25rem; }
    .aj-card__heading-subtitle {
      font-size: 1rem;
      opacity: 0.5;
      font-weight: 400; }
  .aj-card__content {
    font-size: 1rem;
    line-height: 1.5rem;
    padding: 0;
    padding-top: 1rem;
    display: flex;
    justify-content: flex-start;
    flex-direction: column;
    align-items: flex-end; }
    .aj-card__content > p {
      margin-top: 0; }
    .aj-card__content > p:last-of-type {
      margin-bottom: 1.75rem; }
