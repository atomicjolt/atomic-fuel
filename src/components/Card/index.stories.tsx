import React from 'react';
import { Story, Meta } from '@storybook/react';
import StoryWrapper from '../StoryWrapper';
import Card, { Props } from '.';
import { Button, ButtonType } from '../Button';

export default {
  title: 'Card',
  component: Card,
} as Meta;

const Template: Story<Props> = (args) => (
  <StoryWrapper>
    <Card {...args} />
  </StoryWrapper>
);

export const Default = Template.bind({});
Default.args = {
  title: 'Qualtrics Surveys Setup',
  subtitle: 'Step 1 of 2',
  children: (
    <>
      <p>
        In order to use this LTI tool, we need your authorization to integrate
        with Canvas. After authorizing your Canvas account, we'll prompt you to do the
        same with your Qualtrics account.
      </p>
      <Button
        noBold
        buttonType={ButtonType.primary}
      >
        Authorize Canvas
      </Button>
    </>
  ),
};
