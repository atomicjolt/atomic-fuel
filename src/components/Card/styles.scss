.aj-card {
  border-radius: 5px;
  box-shadow: 0px 5px 11px -3px rgba(0, 0, 0, 0.36);
  -webkit-box-shadow: 0px 5px 11px -3px rgba(0, 0, 0, 0.36);
  padding: 1.5rem 2rem;
  max-width: 420px;
  font-family: Lato, sans-serif;
  box-sizing: border-box;
  border: 1px solid rgba(0, 0, 0, 0.036);

  &__heading {
    display: flex;
    flex-direction: column;
    align-items: baseline;
    justify-content: space-between;
    padding-bottom: 0.25rem;

    & > h1,
    & > h2 {
      padding: 0;
      margin: 0;
    }

    &-title {
      font-size: 1.25rem;
    }

    &-subtitle {
      font-size: 1rem;
      opacity: 0.5;
      font-weight: 400;
    }
  }

  &__content {
    font-size: 1rem;
    line-height: 1.5rem;
    padding: 0;
    padding-top: 1rem;
    display: flex;
    justify-content: flex-start;
    flex-direction: column;
    align-items: flex-end;

    & > p {
      margin-top: 0;
    }

    & > p:last-of-type {
      margin-bottom: 1.75rem;
    }
  }
}
