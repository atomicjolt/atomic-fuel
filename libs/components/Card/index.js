"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = Card;

var _react = _interopRequireDefault(require("react"));

var _classnames = _interopRequireDefault(require("classnames"));

require("./styles.css");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function Card(props) {
  var classOverride = props.classOverride,
      classes = props.classes,
      title = props.title,
      subtitle = props.subtitle,
      blank = props.blank,
      children = props.children;
  var baseClass = classOverride || 'aj-card';
  return /*#__PURE__*/_react["default"].createElement("div", {
    className: (0, _classnames["default"])(baseClass, classes)
  }, !blank && /*#__PURE__*/_react["default"].createElement(_react["default"].Fragment, null, /*#__PURE__*/_react["default"].createElement("div", {
    className: "".concat(baseClass, "__heading")
  }, /*#__PURE__*/_react["default"].createElement("h1", {
    className: "".concat(baseClass, "__heading-title")
  }, title), /*#__PURE__*/_react["default"].createElement("h2", {
    className: "".concat(baseClass, "__heading-subtitle")
  }, subtitle)), /*#__PURE__*/_react["default"].createElement("section", {
    className: "".concat(baseClass, "__content")
  }, children)), blank && /*#__PURE__*/_react["default"].createElement(_react["default"].Fragment, null, children));
}