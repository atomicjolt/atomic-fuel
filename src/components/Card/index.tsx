import React from 'react';
import cn from 'classnames';

import './styles.scss';

export interface Props {
  classOverride?: string,
  classes?: string,
  title: string,
  subtitle?: string,
  blank?: boolean,
  children?: React.ReactNode,
}

export default function Card(props: Props) {
  const {
    classOverride,
    classes,
    title,
    subtitle,
    blank,
    children,
  } = props;

  const baseClass = classOverride || 'aj-card';

  return (
    <div className={cn(baseClass, classes)}>
      { !blank &&
        <>
          <div className={`${baseClass}__heading`}>
            <h1 className={`${baseClass}__heading-title`}>{title}</h1>
            <h2 className={`${baseClass}__heading-subtitle`}>{subtitle}</h2>
          </div>
          <section className={`${baseClass}__content`}>
            {children}
          </section>
        </> }
      { blank && <>{children}</> }
    </div>
  );
}
