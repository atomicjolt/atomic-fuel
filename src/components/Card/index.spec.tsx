/**
 * @jest-environment jsdom
 */

import React from 'react';
import { render, screen } from '@testing-library/react';
import Card from '.';

describe('Card component', () => {
  it('Should match default snapshot', () => {
    const { asFragment } = render(
      <Card
        title="Survey Process"
        subtitle="Step 1 of 2"
      />,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('Should not match default snapshot when classOverride prop is added', () => {
    const { asFragment: defaultFragment } = render(
      <Card
        title="Survey Process"
        subtitle="Step 1 of 2"
      />,
    );

    const { asFragment: modifiedFragment } = render(
      <Card
        title="Survey Process"
        subtitle="Step 1 of 2"
        classOverride="aj-huge-card"
      />,
    );

    expect(defaultFragment()).not.toEqual(modifiedFragment());
  });

  it('Should not match default snapshot when classes prop is added', () => {
    const { asFragment: defaultFragment } = render(
      <Card
        title="Survey Process"
        subtitle="Step 1 of 2"
      />,
    );

    const classes = 'material-card a-box';
    const { asFragment: modifiedFragment } = render(
      <Card
        title="Survey Process"
        subtitle="Step 1 of 2"
        classes={classes}
      />,
    );

    expect(defaultFragment()).not.toEqual(modifiedFragment());
  });

  it('Should render children', () => {
    render(
      <Card
        title="Survey Process"
        subtitle="Step 1 of 2"
      >
        <div>恵</div>
      </Card>,
    );

    expect(screen.getByText('恵')).toBeTruthy();
  });

  it('Should not render content if blank property is true', () => {
    const { asFragment } = render(
      <Card
        title="Survey Process"
        subtitle="Step 1 of 2"
        blank
      >
        <div data-testid="grace">恵</div>
      </Card>,
    );

    expect(asFragment()).toMatchSnapshot();
    expect(screen.queryByText(/survey process/i)).toBeFalsy();
    expect(screen.getByTestId('grace')).toBeTruthy();
  });
});
