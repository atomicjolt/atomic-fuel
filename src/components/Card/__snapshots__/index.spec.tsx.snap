// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component Should match default snapshot 1`] = `
<DocumentFragment>
  <div
    class="aj-card"
  >
    <div
      class="aj-card__heading"
    >
      <h1
        class="aj-card__heading-title"
      >
        Survey Process
      </h1>
      <h2
        class="aj-card__heading-subtitle"
      >
        Step 1 of 2
      </h2>
    </div>
    <section
      class="aj-card__content"
    />
  </div>
</DocumentFragment>
`;

exports[`Card component Should not render content if blank property is true 1`] = `
<DocumentFragment>
  <div
    class="aj-card"
  >
    <div
      data-testid="grace"
    >
      ÊÅµ
    </div>
  </div>
</DocumentFragment>
`;
